include/seqan3/core/detail/pack_algorithm.hpp

```diff
@@ -0,0 +1,241 @@
+ * \tparam type_list_t The type list given as a template template parameter.
+ * \tparam pack_t The template arguments contained in `type_list_t` to apply the target function on.
+ *
+ * \details
+ *
+ * The meta algorithms provide a parameter pack version and a seqan3::type_list version.
+ * In the type list version the algorithm is called on the types contained in the enclosing type list after they have
+ * been wrapped in std::type_identity. Thus, the type list version uses the parameter pack version to call the passed
+ * function on the types.
+ * Concrete this helper does the following:
+ *  * expand the types within the type list
+ *  * wrap each type in std::identity and instantiate it (i.e. transform a type into a value)
+ *  * call a target function with the instances of std::identity (as a pack).
+ *
+ * This is a technical trick to make a type representable as a value. Instantiating a type might not always work
+ * because not every type provides a default constructor. In addition it is possible to use incomplete types as well.
+ */
+template <template <typename ...> typename type_list_t, typename ...args_t>
+struct type_list_expander<type_list_t<args_t...>>
+{
+    /*!\brief Invokes the actual function by passing the types as instances of std::type_identity to the target
+     *        function.
+     * \tparam fn_t The type of the function to be called with the expanded list of types.
+     * \param[in] fn The function to be called.
+     * \returns The std::invoke_result of calling `fn` with the expanded list of types.
+     *
+     * \details
+     *
+     * Invokes `fn` by passing the expanded types wrapped in std::type_identity.
+     */
+    template <typename fn_t>
+    //!\cond
+        requires std::invocable<fn_t, std::type_identity<args_t>...>
+    //!\endcond
+    static constexpr std::invoke_result_t<fn_t, std::type_identity<args_t>...> invoke_on_type_identities(fn_t && fn)
+    {
+        return fn(std::type_identity<args_t>{}...);
+    }
+};
+
+//-----------------------------------------------------------------------------
+// all_of
+//-----------------------------------------------------------------------------
+
+/*!\brief Tests whether a given predicate evaluates to `true` for each element in a parameter pack.
+ * \ingroup core
+ *
+ * \tparam unary_predicate_t The function type, like function pointers, functors and lambdas;
+ *                           must model std::predicate expanded on each argument type.
+ * \tparam pack_t The parameter pack of the arguments (each argument type can be different).
+ *
+ * \param[in] fn The predicate to evaluate for every argument.
+ * \param[in] args The parameter pack.
+ *
+ * \details
+ *
+ * This function behaves like std::all_of but on parameter packs. The invocation(s) will be done without any loop.
+ *
+ * ### Example
+ *
+ * \include test/snippet/core/detail/all_of_in_pack.cpp
+ *
+ * ### Complexity
+ *
+ * Linear in the number of elements in the pack.
```

Comment:
```suggestion
 * Linear in the number of types in the pack.
```

Reply:
No, this is about expanding the parameter pack, so it is about elements and not types. Or am I missing something here?

Reply:
right sorry